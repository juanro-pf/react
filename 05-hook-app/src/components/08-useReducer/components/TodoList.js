import React from 'react';
import { TodoListItem } from './TodoListItem';
import PropTypes from 'prop-types';

export const TodoList = ({todos, handleToggle, handleDelete}) => {
    return (
        <ul className='list-group list-group-flush'>
            {
                todos.map( (todo, i) => 
                    <TodoListItem
                        key= {todo.id}
                        todo= {todo}
                        i= {i}
                        handleToggle= {handleToggle}
                        handleDelete= {handleDelete}
                    />
                )
            }
        </ul>
    );
};

TodoList.propTypes= {
    todos: PropTypes.array.isRequired,
    handleToggle: PropTypes.func.isRequired,
    handleDelete: PropTypes.func.isRequired
};