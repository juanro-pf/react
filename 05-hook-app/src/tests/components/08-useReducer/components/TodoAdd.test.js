import React from 'react';

import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import { TodoAdd } from '../../../../components/08-useReducer/components/TodoAdd';

describe('Testing TodoAdd', () => {

    const handleAddTodo= jest.fn();

    const wrapper= shallow(
        <TodoAdd
        handleAddTodo= {handleAddTodo}
        />
    );
   
    test('Snapshot', () => {
        expect(wrapper).toMatchSnapshot(); 
    });

    test('do not call addtodo', () => {
        
        const formSubmit= wrapper.find('form').prop('onSubmit');

        formSubmit({ preventDefault(){} });

        expect(handleAddTodo).not.toHaveBeenCalled();

    });
    
    test('call addtodo', () => {
       
        const value= 'React';
        wrapper.find('input').simulate('change', {
            target: {
                value,
                name: 'description'
            }
        });

        const formSubmit= wrapper.find('form').prop('onSubmit');

        formSubmit({ preventDefault(){} });

        expect(handleAddTodo).toHaveBeenCalledWith({
            desc: value,
            done: false,
            id: expect.any(Number)
        });

        //Por el reset
        expect(wrapper.find('input').prop('value')).toBe('');

    });
    
    
});
