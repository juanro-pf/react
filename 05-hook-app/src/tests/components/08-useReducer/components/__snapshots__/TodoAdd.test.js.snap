// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing TodoAdd Snapshot 1`] = `
<Fragment>
  <h4>
    Add TODO
  </h4>
  <hr />
  <form
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="form-control"
      name="description"
      onChange={[Function]}
      placeholder="learn ..."
      type="text"
      value=""
    />
    <button
      className="btn btn-outline-primary mt-1 btn-block"
      type="submit"
    >
      Add
    </button>
  </form>
</Fragment>
`;
